ops/testing.py

```diff
@@ -2509,38 +2585,54 @@ def make_dir(
             raise pebble.PathError(
                 'generic-file-error',
                 f'permissions not within 0o000 to 0o777: {permissions:#o}')
-        try:
-            self._fs.create_dir(
-                path, make_parents=make_parents, permissions=permissions,
-                user_id=user_id, user=user, group_id=group_id, group=group)
-        except FileNotFoundError as e:
-            # Parent directory doesn't exist and make_parents is False
+        self._check_absolute_path(path)
+        dir_path = self._root / path[1:]
+        if not dir_path.parent.exists() and not make_parents:
             raise pebble.PathError(
-                'not-found', f'parent directory not found: {e.args[0]}')
+                'not-found', f'parent directory not found: {path}')
+        if not dir_path.parent.exists() and make_parents:
+            self.make_dir(
+                os.path.dirname(path),
+                make_parents=True,
+                permissions=permissions,
+                user_id=user_id,
+                user=user,
+                group_id=group_id,
+                group=group)
+        try:
+            permissions = permissions if permissions else 0o755
+            dir_path.mkdir()
+            os.chmod(dir_path, permissions)
+        except FileExistsError:
+            if not make_parents:
+                raise pebble.PathError('generic-file-error', f'mkdir {path}: file exists')
         except NotADirectoryError as e:
             # Attempted to create a subdirectory of a file
             raise pebble.PathError('generic-file-error', f'not a directory: {e.args[0]}')
-        except NonAbsolutePathError as e:
-            raise pebble.PathError(
-                'generic-file-error',
-                f'paths must be absolute, got {e.args[0]!r}'
-            )
 
     def remove_path(self, path: str, *, recursive: bool = False):
         self._check_connection()
-        try:
-            file_or_dir = self._fs.get_path(path)
-        except FileNotFoundError:
+        self._check_absolute_path(path)
+        file_path = self._root / path[1:]
+        if not file_path.exists():
             if recursive:
-                # Pebble doesn't give not-found error when recursive is specified
                 return
-            raise pebble.PathError(
-                'not-found', f'remove {path}: no such file or directory')
-
-        if isinstance(file_or_dir, _Directory) and len(file_or_dir) > 0 and not recursive:
-            raise pebble.PathError(
-                'generic-file-error', 'cannot remove non-empty directory without recursive=True')
-        self._fs.delete_path(path)
+            else:
```

Comment:
Given there's an early return above, I prefer to simplify and avoid the else (similar to how the structure was before).